### PR TITLE
Adds useReducer and refactors the app to search bar interaction

### PR DESCRIPTION
**Pull request by thomedpete**
#### What's this PR do?
This implements the useReducer hook, as well as refactoring how we handle state in app and how pass the results to searchResults.js 

#### Where should the reviewer start?
Check out app then follow the props to nav then searchBar then searchResults

#### How should this be manually tested?
This is tested with cypress

### CODE REVIEW

**Comment by thomedpete** [comment]
Looks like we are failing on circleCI because we are missing a  default case in the reducer  for useReducer

**Comment by Speekins** [comment]
> Looks like we are failing on circleCI because we are missing a default case in the reducer for useReducer

Oh yeah forgot about that. I'm pretty sure you can add `case "default": return {...state}`

**Comment by thomedpete** [comment]
Boom! Added a default case that returns the state as is if none of the other cases are true.  I found a great vid that cleared up some of the confusion I had.

**Comment by bkuhn2** [comment]
Looks great! if there's time tomorrow would someone be able to walk me though how the useReducer hook works?

Also, I was thinking it would make more sense not to have the NavBar on the login page since there's no real 'user' to greet until they log in, and no navigating happening, and also we don't want the person using the site to think they can search for games before they log in. Would be just a matter of having the NavBar component render on all the respective pages apart from login. Let's discuss tomorrow

**Comment by Speekins** [comment]
> Looks great! if there's time tomorrow would someone be able to walk me though how the useReducer hook works?
> 
> Also, I was thinking it would make more sense not to have the NavBar on the login page since there's no real 'user' to greet until they log in, and no navigating happening, and also we don't want the person using the site to think they can search for games before they log in. Would be just a matter of having the NavBar component render on all the respective pages apart from login. Let's discuss tomorrow

Yes, that makes sense. I think `Login` should still be a route, but we can just import `Navbar` into the components that need it to be rendered there.

**Comment by Speekins** [comment]
Thanks for adding tests! Merging right now.

File: cypress/e2e/navbar.spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    cy.get('.welcome-greeting').contains('Welcome,')
+  })
+
+  it('Should have a navbar with a searchbar that can search and redirect to the corect url', () => {
+    cy.get('.search-input')
+      .type('Catan');
+    cy.get('.search-button')
+      .click();
+    cy.url()
+      .should('include', 'http://localhost:3000/search-results/Catan');
```
Comment (letakeane) [inline cypress/e2e/navbar.spec.cy.js:21]: These are excellent tests!

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,21 +1,52 @@
...
+  user: {},
+  userName: '',
+  friendsList: [],
+  gameCollection: [],
+  modalOpen: false,
+  error: null,
+  loading:false
+}
+
+const reducer = (state, action) => {
```
Comment (letakeane) [inline src/components/App/App.js:21]: This is great! If you ever do Redux, you'll see lots of similarities between its reducers and userReducer.

**Pull request merged**